src/components/register/RegisterForm.vue

```diff
@@ -0,0 +1,250 @@
+                required
+                aria-required="true"
+                class="field__input"
+                @focus="resetError()"
+                @blur="check('password')"
+                v-model="password.value"
+            />
+            <p
+                id="password-constraints"
+                class="field__error"
+                v-if="hasError('password')"
+            >
+                Password is required
+            </p>
+        </div>
+        <div class="field">
+            <label class="field__label" for="password2">Confirm password</label>
+            <b-form-input
+                id="password2"
+                type="password2"
+                autocomplete="current-password"
+                aria-describedby="password-constraints"
+                required
+                aria-required="true"
+                class="field__input"
+                @focus="resetError()"
+                @blur="check('password')"
+                v-model="password.value"
+            />
+            <p
+                id="password-constraints"
+                class="field__error"
+                v-if="hasError('password')"
+            >
+                Please, confirm your password
+            </p>
+        </div>
+        <div class="field">
+            <input type="checkbox" id="checkbox" v-model="checked" />
+            <label>
+                I have read and accept the
+                <a
+                    target="_blank"
+                    class="field__action"
+                    href="https://www.plasticorigins.eu/CGU"
+                    >Terms of use and Privacy Policy
+                </a>
+            </label>
+        </div>
+        <button
+            type="submit"
+            class="btn btn-primary mt-6"
+            v-bind:disabled="!isValid || status === 'loading'"
+        >
+            {{ status === "loading" ? "Loading" : "Register" }}
+        </button>
+        <b-card class="bg-danger mt-3" v-if="status === 'error'">
+            <b-card-text>An error occurred. Please try again.</b-card-text>
+        </b-card>
+        <b-card class="bg-success mt-3" v-if="status === 'fulfilled'">
+            <b-card-text>Register in. Please wait...</b-card-text>
+        </b-card>
+    </b-form>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import Component from "vue-class-component";
+@Component({})
+export default class RegistreForm extends Vue {
+    data() {
+        return {
+            email: {
+                error: false,
+                value: ""
+            },
+            password: {
+                error: false,
+                value: ""
+            },
+            name: {
+                error: false,
+                value: ""
+            },
+            name2: {
+                error: false,
+                value: ""
+            },
+            number: {
+                error: false,
+                value: ""
+            },
+            status: "idle"
```

Comment:
On peut utiliser des propriétés de model plus parlante, de ce type
```suggestion
            email: {
                error: false,
                value: ""
            },
            password: {
                error: false,
                value: ""
            },
            firstname: {
                error: false,
                value: ""
            },
            lastname: {
                error: false,
                value: ""
            },
            birthday: {
                error: false,
                value: ""
            },
            status: "idle"
```